src/application/env.rs

```diff
@@ -0,0 +1,181 @@
+#[derive(Debug, Clone)]
+pub struct Environment {
+    pub todo_dir: String,
+    pub todo_file: String,
+    pub done_file: String,
+}
+
+fn initialize_effi_file(file_path: &str) -> Result<(), std::io::Error> {
+    // If a todo.txt or done.txt file does not exist, create an empty one
+
+    let file_path = std::path::Path::new(file_path);
+    if !file_path.exists() {
+        log::debug!(
+            "todo.txt or done.txt does not exist, creating an empty file at {}",
+            file_path.display()
+        );
+        match std::fs::File::create(file_path) {
+            Ok(_) => {
+                return Ok(());
+            }
+            Err(err) => {
+                log::debug!("Unable to create {} because {}", file_path.display(), err);
+                return Err(err);
+            }
+        }
+    }
+    return Ok(());
+}
+
+fn initialize_effi_directory(directory_path: &str) -> Result<(), std::io::Error> {
+    let directory_path = std::path::Path::new(directory_path);
+
+    if !directory_path.exists() {
+        match std::fs::create_dir_all(directory_path) {
+            Ok(_) => {
+                return Ok(());
+            }
+            Err(err) => {
+                log::debug!(
+                    "Unable to create {:?} because {}",
+                    directory_path.display(),
+                    err
+                );
+                return Err(err);
+            }
+        }
+    }
+    return Ok(());
+}
+
+fn get_default_todo_dir() -> Result<String, std::io::Error> {
+    let home_dir = dirs::home_dir();
+    let home_dir_path = match home_dir {
+        Some(home_dir) => home_dir,
+        None => {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "Unable to retrieve the user's home directory",
+            ))
+        }
+    };
+
+    let home_dir_str = match home_dir_path.as_path().to_str() {
+        Some(str) => str,
+        None => {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                "Unable to convert path to string",
+            ))
+        }
+    };
+    return Ok(format!("{home_dir_str}/.local/opt/share/todo"));
+}
+
+impl Environment {
+    pub fn new() -> Result<Self, std::io::Error> {
+        let todo_dir = match std::env::var("TODO_DIR") {
+            Ok(val) => val,
+            Err(_) => {
+                let default_todo_dir = get_default_todo_dir()?;
+                default_todo_dir
+            }
+        };
+
+        let todo_file = format!("{todo_dir}/todo.txt");
+        let done_file = format!("{todo_dir}/done.txt");
+
+        match initialize_effi_directory(&todo_dir) {
```

Comment:
You can use the `?` operator to be more concise.

Reply:
Done, using ```?``` now here